crates/astria-composer/src/searcher/mod.rs

```diff
@@ -376,48 +270,27 @@ impl Searcher {
         Ok(())
     }
 
-    /// Wait until a connection to the sequencer is established.
-    ///
-    /// This function tries to establish a connection to the sequencer by
-    /// querying its `abci_info` RPC. If it fails, it retries for another `n_retries`
-    /// times with exponential backoff.
-    ///
-    /// # Errors
-    ///
-    /// An error is returned if calling the sequencer failed for `n_retries + 1` times.
-    #[instrument(skip_all, fields(
-        retries.max_number = n_retries,
-        retries.initial_delay = %format_duration(delay),
-        retries.exponential_factor = factor,
-    ))]
-    async fn wait_for_sequencer(
-        &self,
-        n_retries: usize,
-        delay: Duration,
-        factor: f32,
-    ) -> eyre::Result<()> {
-        use backon::{
-            ExponentialBuilder,
-            Retryable as _,
-        };
-        debug!("attempting to connect to sequencer",);
-        let backoff = ExponentialBuilder::default()
-            .with_min_delay(delay)
-            .with_factor(factor)
-            .with_max_times(n_retries);
-        (|| {
-            let client = self.sequencer_client.clone();
-            async move { client.abci_info().await }
-        })
-        .retry(&backoff)
-        .notify(|err, dur| warn!(error.msg = %err, retry_in = %format_duration(dur), "failed getting abci info; retrying"))
+    async fn wait_for_executor(&self) -> eyre::Result<()> {
+        // wait to receive executor status
+        let mut status = self.executor_status.clone();
+        async move {
+            match status.wait_for(executor::Status::is_connected).await {
```

Comment:
I don't think we need a match statement for this - just upwards propagate the error and then return and `Ok(())`

Reply:
This was extracted from waiting for the collectors. I think (but am not sure) that it was necessary to do it there explicitly to avoid any sort of implicit `+ 'a` bound being added to the returned future.

I agree that in this case the entire `async move` is probably not necessary